.github/workflows/publish-snapshot.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish Snapshot
+
+on:
+  schedule:
+  - cron: '0 0 * * *'
+
+jobs:
+  publish-snapshot:
+    if: ${{ startsWith(github.repository, 'apache/') }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        branch:
+          - master
+          - branch-1.3
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@master
+      with:
+        ref: ${{ matrix.branch }}
+    - name: Cache Maven local repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: snapshot-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          snapshot-maven-
+    - name: Install Java 8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - name: Publish snapshot
+      env:
+        ASF_USERNAME: ${{ secrets.NEXUS_USER }}
+        ASF_PASSWORD: ${{ secrets.NEXUS_PW }}
+      run: ./build/mvn clean deploy -DskipTests -Pspark-provided -s ./build/release/asf-settings.xml
```

Comment:
shall we rm all the kyuubi artifacts from cache？

Reply:
Updated.

Reply:
I’AM OK to publish it as long as the snapshot has nothing to do with final releases

Reply:
Just realized that Spark 3.2 will support both Scala 2.12 and 2.13. But `kyuubi-extension-spark-3-1` only build against Scala 2.12, and the package name lack the `_2.12` suffix, it may confuse users who want to use it with Spark 3.2. (I guess only few users will run Spark 3.2 with Scala 2.13 in production?)